File: Scripts/Common/Revision.rb
#
# The MIT License
#
# Copyright (c) 2019 Volodymyr Gorlov (https://github.com/vgorloff)
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.
#

class Revision

   # See also: $ToolChain/Sources/swift/utils/update_checkout/update-checkout-config.json

   def self.swift
      return "443e9a41d1a7c1fcae280490144fcbf8461d3499"
      # 1.0.54 @ 2020.03.24 @ swift-5.2-RELEASE: 443e9a41d1a7c1fcae280490144fcbf8461d3499
      # 1.0.53 @ swift-DEVELOPMENT-SNAPSHOT-2020-01-13-a: 3b9014dc9acae3d95255a83befd85b52e7176acb
      # 1.0.51 @ swift-DEVELOPMENT-SNAPSHOT-2019-12-27-a: 746b58e8e1c1ef1ac09d8031875fd2a08b65597c
      # 1.0.50 @ swift-DEVELOPMENT-SNAPSHOT-2019-10-14-a: 9a4abf47a450aeff6a3231225fd2cf18cc7a940a
      # 1.0.50 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-26-a: 8a91e9802448de399df4ee3ed24c63dc25af7e67
      # 1.0.50 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-18-a: 4dac0d1e8bffee79728f04784961b3bc26ed9dbc
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-16-a: f302da0f9b856592992206afa75dd95668a192e1
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-08-30-a: dcd0893efe43639dddf6fc1637022822080e6a6a
   end

   def self.dispatch
      return "c992dacf3ca114806e6ac9ffc9113b19255be9fe"
      # 1.0.52 @ swift-DEVELOPMENT-SNAPSHOT-2020-01-13-a: c992dacf3ca114806e6ac9ffc9113b19255be9fe
      # 1.0.52 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-29-a: 2accb0b97738b9ba9a0ffc1cb335716b70f9f65a
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-26-a: 970d562fcaee756bae9d20386e64bba90e635818
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-16-a: c023edd1cd02720407c62f4fac6a3f2c92cd0069
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-02-a: 90a45ce22f62c7c380339aa847dae4ab2d66ee30
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-08-22-a: 3da29ddcab09e4d09b015624e377f73a82c8dc9c
   end

   def self.foundation
      return "62809e9e2f918e6c11e47cd6a8d858429c77b743"
      # 1.0.53 @ swift-DEVELOPMENT-SNAPSHOT-2020-01-05-a: 62809e9e2f918e6c11e47cd6a8d858429c77b743
      # 1.0.53 @ swift-DEVELOPMENT-SNAPSHOT-2019-12-27-a: 86eec6148c1e4ef1c56dbd5e53c5cbf6d023f202
      # 1.0.53 @ swift-DEVELOPMENT-SNAPSHOT-2019-11-30-a: 2195121bf063e2db7c71080fcae7e76e8a43950e
      # 1.0.53 @ swift-DEVELOPMENT-SNAPSHOT-2019-11-06-a: c093eb65ed3aabb334857714e01989ecfb875120
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-11-01-a: facf5715d25d1765f10e4319b0701f300d9248e0
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-10-30-a: 5bb98287f5945a8cbb37df5a69741328488d2dd0
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-30-a: f62c026cb2f2f162bb3b3d93ae49969b60652e0b
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-04-a: c9c34a9447682131ae49332e3d2a20b8c82ca01e
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-09-02-a: 3e5366251756d7ffcf289c0faff865ca7dd23417
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-08-28-a: d99abed49a0763eefb7e1e1cdaa9654fb38c7c1b
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-08-16-a: 2e4734c674247b011a67996a9472f15ecb394e5e
      # 1.0.49 @ swift-DEVELOPMENT-SNAPSHOT-2019-08-14-a: f2d055cf77f59568c23412b2179e72193abac052
   end

   def self.spm
      # master @ 1.0.43 3dad9e7cfe20d1b48256afca13df293bdca870b9
      # swift-5.0-branch @ 1.0.43 3a57975e10be0b1a8b87992ddf3a49701036f96c
      # swift-5.1-branch @ 1.0.37 0996526dbf2d5adbdef4b133312e441c0deb3db7
      return "3dad9e7cfe20d1b48256afca13df293bdca870b9"
   end

   def self.llb
      # master @ 1.0.43 5bf016632c0e85d835c63b84b43d3c887f01418f
      # swift-5.0-branch @ 1.0.43 4f77fc47e2f5fcddc057abc8391f7c23d1e8d275
      # swift-5.1-branch @ 1.0.37 f73b84bc1525998e5e267f9d830c1411487ac65e
      return "5bf016632c0e85d835c63b84b43d3c887f01418f"
   end

   def self.llvm
      return "c2133f2b08047708b17230b598cb789062143650"
      # 1.0.54 @ 2020.03.23 @ swift-5.2-RELEASE: c2133f2b08047708b17230b598cb789062143650
      # 1.0.54 @ swift-5.1.5-RELEASE: 198fa1e3efcce2fd302f7aaa077aec443a82394b
      # 1.0.51 @ swift-DEVELOPMENT-SNAPSHOT-2020-01-03-a: 019936b2100dc33c8946928d424b5cff4211acea
   end

   def self.cmark
      return "bfa95d55b535fa178f75484b5e8f82ae3d8517af"
      # 1.0.54 @ 2019.10.10 @ swift-5.2-RELEASE: bfa95d55b535fa178f75484b5e8f82ae3d8517af
      # 1.0.51 @ swift-DEVELOPMENT-SNAPSHOT-2020-01-03-a: 31e12547c4de3dfcacc69855d93298cb90fd4f27
      # 1.0.50 @ swift-DEVELOPMENT-SNAPSHOT-2019-10-14-a: bfa95d55b535fa178f75484b5e8f82ae3d8517af
      # 1.0.49 @ swift-5.1.1-RELEASE: 32fa49671d0fc5d1f65d2bcbabfb1511a9d65c27
   end

   def self.ssl
      return "894da2fb7ed5d314ee5c2fc9fd2d9b8b74111596"
      # 1.0.53 @ OpenSSL_1_1_1d: 894da2fb7ed5d314ee5c2fc9fd2d9b8b74111596
      # OpenSSL_1_1_1c: 97ace46e11dba4c4c2b7cb67140b6ec152cfaaf4 (1.0.26)
      # OpenSSL_1_1_1b: 50eaac9f3337667259de725451f201e784599687
      # OpenSSL_1_1_1a: d1c28d791a7391a8dc101713cd8646df96491d03
   end

   def self.icu
      return "e2d85306162d3a0691b070b4f0a73e4012433444"
      # 1.0.26 @ v64-2: e2d85306162d3a0691b070b4f0a73e4012433444
      # v64-rc2: 67d218f2476ac543de2ed843fa080892972c604a
   end

   def self.curl
      return "2cfac302fbeec68f1727cba3d1705e16f02220ad"
      # 1.0.53 @ curl-7_68_0: 2cfac302fbeec68f1727cba3d1705e16f02220ad
      # 1.0.26 @ 7_65_1: 69248b58f649e35b09a126c12781353e3471f5c6
   end

   def self.xml
      # v2.9.9: f8a8c1f59db355b46962577e7b74f1a1e8149dc6 (1.0.26)
      return "f8a8c1f59db355b46962577e7b74f1a1e8149dc6"
   end

end

File: lib/Git/Revision.js
/*
 * The MIT License
 *
 * Copyright (c) 2019 Volodymyr Gorlov (https://github.com/vgorloff)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

// See also: $ToolChain/Sources/swift/utils/update_checkout/update-checkout-config.json

module.exports = {
  // 1.0.77 - 4c45f525dc315a618868caae7d976d95359700a4 - tag:swift-5.5.2-RELEASE
  // 1.0.76 - 11a4d5a4f54cdfb803b3c75e3c4dd5d2dfc1974c - tag:swift-5.5.1-RELEASE
  // 1.0.74 - d2054e1c741f549caffbe1f286a9b11066b5c696 - tag:swift-5.5-RELEASE
  // 1.0.73 - 282fe25d1757ff9974ade028d92111acdae6876a - tag:swift-5.4.3-RELEASE
  swift: "4c45f525dc315a618868caae7d976d95359700a4",

  // 1.0.77 - 5cc1c6679822fd18368664482a5bba57270467f4 - tag:swift-5.5.2-RELEASE
  // 1.0.76 - e7c11bcb69c785c3470c91daf9c5d1b13162e7a6 - tag:swift-5.5.1-RELEASE
  // 1.0.74 - 851fbd3cb89385ba733bc37d6149280996715898 - tag:swift-5.5-RELEASE
  // 1.0.73 - f13ea5dcc055e5d2d7c02e90d8c9907ca9dc72e1 - tag:swift-5.4.3-RELEASE
  dispatch: "5cc1c6679822fd18368664482a5bba57270467f4",

  // 1.0.77 - 491a217baa8a0ef4ae9c8603c434c069cc090c50 - tag:swift-5.5.2-RELEASE
  // 1.0.74 - 5663ff1147d8f162f1cb44bfdb3e8c42addb7b69 - tag:swift-5.5-RELEASE
  // 1.0.73 - dee1e194a44e660c205960adb470ea0981f78b49 - tag:swift-5.4.3-RELEASE
  foundation: "491a217baa8a0ef4ae9c8603c434c069cc090c50",

  // 1.0.77 - c315411fac10439aa7a09dce976b5716af9f149c - tag:swift-5.5.2-RELEASE
  // 1.0.76 - 47e0531f53ce4508b522a28d407ac39b639781f4 - tag:swift-5.5.1-RELEASE
  // 1.0.74 - fbf00809f11885f819457563beb069a2298b368f - tag:swift-5.5-RELEASE
  // 1.0.73 - 8ee67e3af469eb38c9616f971abd5a47fe8d6405 - tag:swift-5.4.3-RELEASE
  llvm: "c315411fac10439aa7a09dce976b5716af9f149c",

  // 1.0.77 - 9c8096a23f44794bde297452d87c455fc4f76d42 - tag:swift-5.5.2-RELEASE
  // 1.0.74 - 9c8096a23f44794bde297452d87c455fc4f76d42 - tag:swift-5.5-RELEASE
  // 1.0.73 - 9c8096a23f44794bde297452d87c455fc4f76d42 - tag:swift-5.4.3-RELEASE
  cmark: "9c8096a23f44794bde297452d87c455fc4f76d42",

  // 1.0.77 - 7a1f113534689c77b3a4110288478580b7b8d91c - tag:swift-5.5.2-RELEASE
  // 1.0.74 - 519aa4a70db7d49e894ed83ea61d2a4d24c098f0 - tag:swift-5.5-RELEASE
  // 1.0.73 - 374cdb42d7b146837de70489a04c5df956092685 - tag:swift-5.4.3-RELEASE
  spm: "7a1f113534689c77b3a4110288478580b7b8d91c",

  // 1.0.77 - 83c4bcb8dfca48cc065325287b55d08ff7b26428 - tag:swift-5.5.2-RELEASE
  // 1.0.74 - b5d9b4a9995c05688ae5f3b87a0d7ac0dc45c6c6 - tag:swift-5.5-RELEASE
  // 1.0.73 - eb56a00ed9dfd62c2ce4ec86183ff0bc0afda997 - tag:swift-5.4.3-RELEASE
  llb: "83c4bcb8dfca48cc065325287b55d08ff7b26428",

  // 1.0.77 - 3b586ce12865db205081acdcea79fe5509b28152 - tag:swift-5.5.2-RELEASE
  // 1.0.74 - 1b21e2ce36891ed4f458421a83b5d9e886acd4cd - tag:swift-5.5-RELEASE
  // 1.0.73 - d7bd4375c26e7dab2c17791cfa06f9b981d02339 - tag:swift-5.4.3-RELEASE
  tsc: "3b586ce12865db205081acdcea79fe5509b28152",

  // 1.0.77 - 86c54dacd270e0c43374c0cb9b2ceb2924c9ea72 - tag:swift-5.5.2-RELEASE
  // 1.0.74 - 12cd468c8182b4e1038268c664354005803fcdef - tag:swift-5.5-RELEASE
  // 1.0.73 - 93e8b927225a62b963ebe13ab11e04192fa8a67b - tag:swift-5.4.3-RELEASE
  sd: "86c54dacd270e0c43374c0cb9b2ceb2924c9ea72",

  // 1.0.71 - 986d191f94cec88f6350056da59c2e59e83d1229 - tag:0.4.3
  // 1.0.70 - 986d191f94cec88f6350056da59c2e59e83d1229 - tag:0.4.3
  // 1.0.69 - 831ed5e860a70e745bc1337830af4786b2576881 - tag:0.4.1
  sap: "986d191f94cec88f6350056da59c2e59e83d1229",

  // 1.0.71 - 9ff1cc9327586db4e0c8f46f064b6a82ec1566fa - tag:4.0.6
  // 1.0.70 - 9ff1cc9327586db4e0c8f46f064b6a82ec1566fa - tag:4.0.6
  // 1.0.69 - 81a65c4069c28011ee432f2858ba0de49b086677 - tag:3.0.1
  yams: "9ff1cc9327586db4e0c8f46f064b6a82ec1566fa",

  // 1.0.74 - 0141f53dd525706c803b0c20aa8ad36f9ecd45e5 - tag:1.1.5
  sc: "0141f53dd525706c803b0c20aa8ad36f9ecd45e5",

  // 1.0.53 - 894da2fb7ed5d314ee5c2fc9fd2d9b8b74111596 - tag:OpenSSL_1_1_1d @ 10. September 2019 at 15:13:07 CEST
  ssl: "894da2fb7ed5d314ee5c2fc9fd2d9b8b74111596",

  // 1.0.26 - e2d85306162d3a0691b070b4f0a73e4012433444 - tag:v64-2 @ 17. April 2019 at 17:58:08 CEST
  icu: "e2d85306162d3a0691b070b4f0a73e4012433444",

  // 1.0.67 - 566b74a0e19b9aa610f4931e5bfd339bcf8e9147 - tag:curl-7_76_1 @ 14. April 2021 at 07:56:23 CEST
  // 1.0.53 - 2cfac302fbeec68f1727cba3d1705e16f02220ad - tag:curl-7_68_0 @ 4. January 2020 at 22:48:15 CET
  curl: "566b74a0e19b9aa610f4931e5bfd339bcf8e9147",

  // 1.0.26 - f8a8c1f59db355b46962577e7b74f1a1e8149dc6 - tag:v2.9.9 @ 3. January 2019 at 19:14:17 CET
  xml: "f8a8c1f59db355b46962577e7b74f1a1e8149dc6",
};
